main.js

```diff
@@ -0,0 +1,142 @@
+// The app collection
+const AppCollection = Backbone.Collection.extend({
+  // grabs the url thats passed down from the input
+  url() {
+    return this.models[0].attributes.url;
+  },
+  // on the first instance on the add event do the api fetch
+  initialize() {
+    this.on('add', function (model) {
+      model.fetch();
```

Comment:
You would never want to do this - fetch on add? It would get added after it's fetched, so I'm not sure what this would do?